### DISCUSSION TITLE
widget.onClick receiving empty targetId with nested tags

### DISCUSSION THREAD

**Discussion by jacopo-degattis** [johnlindquist/kit#865] [Errors]
Hello everyone,
I'm currently trying out script-kit and I was experimenting button's click events.
I noticed that, when I have a nested tag such as this `<button id='test'><pre>CLICK</pre></button>` the callback function will receive an empty targetId while if I use a not nested tag such as `<button id='test'>CLICK</button>` it all works fine.
I wonder what the error could be.
Any help is much appreciated
Thanks in advance

**Comment by johnlindquist** [comment]
I'll dig into it. I think it's just sending back the I'd of the element directly clicked.

I'll see if I can make it so it looks for the closest parent id if one doesn't exist on the element.

**Reply by jacopo-degattis** [reply]
Right I see, let me know If you have any news.
Thank you !

**Reply by johnlindquist** [reply]
Also, if you prefer to real-time chat, we're starting up a Discord. Would love your feedback :)

https://discord.gg/qcWXkZJC
